desktop/libs/indexer/src/indexer/indexers/sql.py

```diff
@@ -310,25 +321,76 @@ def create_table_from_local_file(self, source, destination, start_time=-1):
               'columns': ',\n'.join(['  `%(name)s` %(type)s' % col for col in columns]),
             }
 
+    elif editor_type == 'impala':
+      sql = '''CREATE TABLE IF NOT EXISTS %(database)s.%(table_name)s_tmp (
+%(columns)s);
+      ''' % {
+                'database': database,
+                'table_name': table_name,
+                'columns': ',\n'.join(['  `%(name)s` string' % col for col in columns]),
+            }                                                 # Impala does not implicitly cast between string and numeric or Boolean types.
+
     path = urllib_unquote(source['path'])
 
-    if path:                                                     # data insertion
+    if path:                                                  # data insertion
+
       with open(BASE_DIR + path, 'r') as local_file:
         reader = csv.reader(local_file)
-        list_of_tuples = list(map(tuple, reader))
-
-        if source['format']['hasHeader']:
-          list_of_tuples = list_of_tuples[1:]
-
-        csv_rows = str(list_of_tuples)[1:-1]
-
-        if editor_type in ('hive', 'mysql'):
-          sql += '''\nINSERT INTO %(database)s.%(table_name)s VALUES %(csv_rows)s;
-          '''% {
+        csv_rows = ''
+
+        for count, row in enumerate(reader):
+          if source['format']['hasHeader'] and count == 0:
+            continue
+          if editor_type == 'impala':                         # for the boolean col updating csv_val to (1,0)
+            row = self.bool_col_update(row, columns)
+          csv_rows += (str(tuple(row)) + ', ')
+
+        if csv_rows:
+          csv_rows = csv_rows[:-2]
+
+          if editor_type in ('hive', 'mysql'):
+            sql += '''\nINSERT INTO %(database)s.%(table_name)s VALUES %(csv_rows)s;
+            '''% {
+                    'database': database,
+                    'table_name': table_name,
+                    'csv_rows': csv_rows
+                  }
+          elif editor_type == 'impala':
+            sql += '''\nINSERT INTO %(database)s.%(table_name)s_tmp VALUES %(csv_rows)s;
+            '''% {
+                    'database': database,
+                    'table_name': table_name,
+                    'csv_rows': csv_rows
+                  }
```

Comment:
Any chance to refactor to avoid all the repeated patterns?

e.g. at this end'

```
sql += load_create_template % 
{
      'database': database,
      'table_name': table_name,
      'csv_rows': csv_rows
    }
```